Upstream-Status: Backport [https://github.com/OP-TEE/optee_os/pull/4000]
Signed-off-by: Ross Burton <ross.burton@arm.com>

From 79384a244d1972cdb02114f5cde6fcaf2e014c9b Mon Sep 17 00:00:00 2001
From: Jerome Forissier <jerome@forissier.org>
Date: Tue, 28 Jul 2020 17:28:43 +0200
Subject: [PATCH] build: fix race when generating conf.mk

This patch fixes the following error triggered by a heavily parallel build:

 echo sm := ta_arm64 > .../export-ta_arm64/mk/conf.mk.tmp
 /bin/bash: .../export-ta_arm64/mk/conf.mk.tmp: No such file or directory

Fixes: https://github.com/OP-TEE/optee_os/issues/3999
Signed-off-by: Jerome Forissier <jerome@forissier.org>
---
 ta/ta.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ta/ta.mk b/ta/ta.mk
index 59ed87f71d..ad97edaf08 100644
--- a/ta/ta.mk
+++ b/ta/ta.mk
@@ -190,6 +190,7 @@ define mk-file-export
 .PHONY: $(conf-mk-file-export)
 $(conf-mk-file-export):
 	@$$(cmd-echo-silent) '  CHK    ' $$@
+	$(q)mkdir -p $$(dir $$@)
 	$(q)echo sm := $$(sm-$(conf-mk-file-export)) > $$@.tmp
 	$(q)echo sm-$$(sm-$(conf-mk-file-export)) := y >> $$@.tmp
 	$(q)($$(foreach v, $$(ta-mk-file-export-vars-$$(sm-$(conf-mk-file-export))), \
